src/internal/streamstore/details.go

```diff
@@ -0,0 +1,159 @@
+package streamstore
```

Comment:
Moved here from streamstore.go.  Some changes involved, a bit of code tidying, and then compliance with the new interfaces.

Reply:
Decided to split that into its own PR.  See https://github.com/alcionai/corso/pull/2725 first.